### PR TITLE
Dat 3638b

### PR DESCRIPTION
**Pull request by rybmat**
maintanance script for migrating hero module data to curated content

### CODE REVIEW

File: includes/wikia/services/CommunityDataService.class.php [hunk 1]
```diff
@@ -93,6 +93,10 @@ public function getCommunityImageId() {
 		return isset( $data[ 'image_id' ] ) ? $data[ 'image_id' ] : 0;
 	}
```
Comment (sebastianmarzjan) [inline includes/wikia/services/CommunityDataService.class.php:95]: Please add a docblock describing the structure of returned value

File: extensions/wikia/CuratedContent/maintenance/HeroImageToCuratedContentMigrator.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+		$wikiData = new WikiDataModel(Title::newMainPage()->getText());
+		$wikiData->getFromProps();
+
+		// migrate description if there is no description in community data
+		if (empty($commData->getCommunityDescription()) && !empty($wikiData->description)) {
+			$curatedData['community_data']['description'] = $wikiData->description;
+		}
+
+		$heroImage = Title::newFromText($wikiData->getImageName(), NS_IMAGE);
```
Comment (sebastianmarzjan) [inline extensions/wikia/CuratedContent/maintenance/HeroImageToCuratedContentMigrator.php]: Is NS_IMAGE a key namespace or an alias to NS_FILE ? If so, then please use the 'canonical' one :)

File: extensions/wikia/CuratedContent/maintenance/HeroImageToCuratedContentMigrator.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		// migrate image if there is no image in community data
+		if (empty($commData->getCommunityImageId()) && !empty($heroImage)) {
+			$curatedData['community_data']['image_id'] = $heroImage->getArticleID();
+		}
+
+		$commData->setCuratedContent($curatedData);
```
Comment (sebastianmarzjan) [inline extensions/wikia/CuratedContent/maintenance/HeroImageToCuratedContentMigrator.php]: Please add some message to logging; WikiFactory does that by default ( see WikiFactory::setVarById ), but we'd like to have a reason passed to easily filter out script-changed results. Also, a dry-run option would be helpful for testing.

File: extensions/wikia/CuratedContent/maintenance/HeroImageToCuratedContentMigrator.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+		}
+
+		$heroImage = Title::newFromText($wikiData->getImageName(), NS_FILE);
+		// migrate image if there is no image in community data
+		if (empty($commData->getCommunityImageId()) && !empty($heroImage)) {
+			$curatedData['community_data']['image_id'] = $heroImage->getArticleID();
+		}
+
+		if ($dryRun) {
+			$this->output("\ncurated content after migration:\n".json_encode($curatedData)."\n");
```
Comment (sebastianmarzjan) [inline extensions/wikia/CuratedContent/maintenance/HeroImageToCuratedContentMigrator.php:43]: Before would be also good :)

**Comment by sebastianmarzjan** [comment]
One more suggestion and LG.

File: extensions/wikia/CuratedContent/maintenance/HeroImageToCuratedContentMigrator.php (hunk 4 above)
Comment (rybmat) [inline extensions/wikia/CuratedContent/maintenance/HeroImageToCuratedContentMigrator.php:43]: https://github.com/Wikia/app/blob/DAT-3638b/extensions/wikia/CuratedContent/maintenance/HeroImageToCuratedContentMigrator.php#L25

Comment (rybmat) [inline extensions/wikia/CuratedContent/maintenance/HeroImageToCuratedContentMigrator.php:43]: ;)

Comment (sebastianmarzjan) [inline extensions/wikia/CuratedContent/maintenance/HeroImageToCuratedContentMigrator.php:43]: :+1: - missed this one :)

**Comment by sebastianmarzjan** [comment]
LGTM

**Pull request merged**